java/src/main/java/io/getconnect/client/java/JavaConnectClient.java

```diff
@@ -21,4 +24,14 @@ public JavaConnectClient(String projectId, String apiKey, String eventStoreDir)
     private JavaConnectClient(String projectId, String apiKey, String eventStoreDir, JsonSerializer serializer) throws IOException {
         super(projectId, apiKey, null, new UrlConnectionHttpClient(), serializer, new FileEventStore(projectId, new File(eventStoreDir), serializer));
     }
+
+    /**
+     * Encrypt filtered key for use with the Connect API.
+     * @param key the definition of the filtered key.
+     * @param masterKey the master key for the Connect project
+     * @return The encrypted filtered key.
+     */
+    public static String generateFilteredKey(final Map<String, Object> key, String masterKey) throws FilteredKeyException {
+        return FilteredKey.encrypt(new JacksonJsonSerializer(), key, masterKey);

```

Comment:
We should use the serializer that is stored from the constructor so it can be overriden as needed.